pyk/src/pyk/proof/reachability.py

```diff
@@ -793,14 +795,24 @@ def step_proof(self, step: APRProofStep) -> list[APRProofResult]:
         if step.circularity and not step.nonzero_depth and (execute_depth is None or execute_depth > 1):
             execute_depth = 1
 
-        extend_result = self.kcfg_explore.extend_cterm(
-            step.node.cterm,
-            execute_depth=execute_depth,
-            cut_point_rules=cut_rules,
-            terminal_rules=self.terminal_rules,
-            module_name=step.module_name,
-            node_id=step.node.id,
-        )
+        if step.node.id in self.next_steps:
+            extend_results = [self.next_steps.pop(step.node.id)]
+        else:
+            extend_results = self.kcfg_explore.extend_cterm(
+                step.node.cterm,
+                execute_depth=execute_depth,
+                cut_point_rules=cut_rules,
+                terminal_rules=self.terminal_rules,
+                module_name=step.module_name,
+                node_id=step.node.id,
+            )
+
+        assert len(extend_results) == 1 or len(extend_results) == 2
+        extend_result = extend_results[0]
+        if len(extend_results) == 2:
+            assert step.node.id not in self.next_steps
+            self.next_steps[step.node.id] = extend_results[1]
```

Comment:
The problem is here - this is wrong and I'm not sure how to fix it - I need the identifier of the node that is to be created by the first step, but I don't have it. I know what it is, it is the `_node_id` of the associated KCFG, but the KCFG is not here.

Alternatively, I could look for the predecessors of the node, but I still don't have the KCFG here.

Alternatively, I could try to move this logic into the KCFG logic, but I'm neither sure how nor that it's the way to go.